File: test/fixtures/ruby/post_with_data_binary.rb
require 'net/http'

uri = URI('http://localhost:28139/post')
req = Net::HTTP::Post.new(uri)
req.content_type = 'application/x-www-form-urlencoded'

req.body = '{"title":"china1"}'

req_options = {
  use_ssl: uri.scheme == 'https'
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end

File: test/fixtures/kotlin/post_with_data_binary.kt
import java.io.IOException
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody

val client = OkHttpClient()

val MEDIA_TYPE = "application/x-www-form-urlencoded".toMediaType()

val requestBody = "{\"title\":\"china1\"}"

val request = Request.Builder()
  .url("http://localhost:28139/post")
  .post(requestBody.toRequestBody(MEDIA_TYPE))
  .header("Content-Type", "application/x-www-form-urlencoded")
  .build()

client.newCall(request).execute().use { response ->
  if (!response.isSuccessful) throw IOException("Unexpected code $response")
  response.body!!.string()
}
